### ISSUE TITLE
Shared database is reloading constantly causing UI lockup

### ISSUE THREAD

**Issue by chrisovergaauw** [keepassxreboot/keepassxc#4261]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )

## Expected Behavior
open app, unlock db. nothing else should happen.

## Current Behavior
open app, unlock db. UI starts blinking within a minute.
Every blink, the save button disables as well as the list of secrets.
app becomes basically unusable.

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
1. open app
2. unlock db
3. enable Automatically save after every change.

## Context
1 local db. 2 shared db's via google cloud.

## Debug Info
KeePassXC - Version 2.5.3
Revision: f8c962b

Qt 5.14.0
Debugging mode is disabled.

Operating system: macOS Mojave (10.14)
CPU architecture: x86_64
Kernel: darwin 18.7.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
 libgcrypt 1.8.5

Libraries:
- LIBS

Operating system: OS
CPU architecture: ARCH
Kernel: KERNEL

Enabled extensions:
- EXTENSIONS

**Comment by droidmonkey** [comment]
Something is modifying your database file constantly. Can you clarify your context? Are you storing a database locally and on Google cloud?

**Comment by chrisovergaauw** [comment]
two of the databases are stored on Google cloud and shared with other people.
I recently upgraded my version. Could it be caused by a "mismatching version"? If one of them had the app open as well, on possibly an older version? (I'll ask them tomorrow and investigate)

I would hope that upgrades should not be able to cause this because it is getting more difficult to get everyone on the exact same version all the time.

**Comment by droidmonkey** [comment]
2.5.2 and 2.5.3 versions introduced improved behavior with shared databases. Specifically in 2.5.2 we will not try to reload a database that has not changed on disk even though the modified time changes. Please do some more testing to see where things are breaking down.

**Comment by adoering-te** [comment]
I think I have this happening to me on both Windows and macOS. My DB file is stored on Google's Shared/Team Drives. I have the copies of the DB file saved locally offline. 

Most recently, I attempted to save my DB, and the UI locked up to an all gray screen and nothing responds, and the new details that are trying to be saved do not save. 

![image](https://user-images.githubusercontent.com/48135402/76650220-5c96ce00-651f-11ea-9bb4-98fcc4c5a191.png)

I received this after I clicked File > Close Database/Quit
